core/src/main/java/com/linecorp/armeria/client/retry/RetryRuleBuilder.java

```diff
@@ -0,0 +1,282 @@
+    }
+
+    /**
+     * Adds the specified {@link HttpStatus}es for a {@link RetryRule} which will retry
+     * if the response status is one of the specified {@link HttpStatus}es.
+     */
+    public RetryRuleBuilder onStatus(Iterable<HttpStatus> statuses) {
+        requireNonNull(statuses, "statuses");
+        checkArgument(!Iterables.isEmpty(statuses), "statuses can't be empty");
+
+        statusesBuilder.addAll(statuses);
+        return this;
+    }
+
+    /**
+     * Adds the specified {@code statusFilter} for a {@link RetryRule} which will retry
+     * if the response status matches the specified {@code statusFilter}.
+     */
+    public RetryRuleBuilder onStatus(Predicate<? super HttpStatus> statusFilter) {
+        requireNonNull(statusFilter, "statusFilter");
+        if (this.statusFilter != null) {
+            this.statusFilter = this.statusFilter.or(statusFilter);
+        } else {
+            @SuppressWarnings("unchecked")
+            final Predicate<HttpStatus> cast = (Predicate<HttpStatus>) statusFilter;
+            this.statusFilter = cast;
+        }
+        return this;
+    }
+
+    /**
+     * Adds the specified exception type for a {@link RetryRule} which will retry
+     * if an {@link Exception} is raised and that is instance of the specified {@code exception}.
+     */
+    public RetryRuleBuilder onException(Class<? extends Throwable> exception) {
+        requireNonNull(exception, "exception");
+        return onException(exception::isInstance);
+    }
+
+    /**
+     * Adds the specified {@code exceptionFilter} for a {@link RetryRule} which will retry
+     * if an {@link Exception} is raised and the specified {@code exceptionFilter} returns {@code true}.
+     */
+    public RetryRuleBuilder onException(Predicate<? super Throwable> exceptionFilter) {
+        requireNonNull(exceptionFilter, "exceptionFilter");
+        if (this.exceptionFilter != null) {
+            this.exceptionFilter = this.exceptionFilter.or(exceptionFilter);
+        } else {
+            @SuppressWarnings("unchecked")
+            final Predicate<Throwable> cast = (Predicate<Throwable>) exceptionFilter;
+            this.exceptionFilter = cast;
+        }
+        return this;
+    }
+
+    /**
+     * Makes a {@link RetryRule} retry on any {@link Exception}.
+     */
+    public RetryRuleBuilder onException() {
+        return onException(unused -> true);
+    }
+
+    /**
+     * Makes a {@link RetryRule} retry on an {@link UnprocessedRequestException} which means that the request
+     * has not been processed by the server. Therefore, you can safely retry the request without worrying about
+     * the idempotency of the request.
+     */
+    public RetryRuleBuilder onUnprocessed() {
+        return onException(UnprocessedRequestException.class);
+    }
+
+    /**
+     * Sets the {@linkplain Backoff#ofDefault() default backoff} and returns a newly created {@link RetryRule}.
+     */
+    public RetryRule thenBackoff() {
```

Comment:
Perhaps `thenDefaultBackoff`?

Reply:
It was `thenDefaultBackoff()` originally. but changed here https://github.com/line/armeria/pull/2661#discussion_r414266200

Reply:
Oh I missed that. 😅